### DISCUSSION TITLE
{Network}Queries['{pallet}']['{call}'] from @polkadot-api/descriptors is returning a slightly different type than what I'd expect

### DISCUSSION THREAD

**Discussion by Tbaut** [polkadot-api/polkadot-api#818] [Q&A] [answered]
This is probably a dumb question, but I didn't figure it out. If I want to strongly type a variable with the return type of a query, I thought I could use `PaseoQueries['Multisig']['Multisigs']` for instance, which yields

```ts
{
    KeyArgs: [SS58String, FixedSizeBinary<32>];
    Value: {
        when: {
            height: number;
            index: number;
        };
        deposit: bigint;
        depositor: SS58String;
        approvals: SS58String[];
    };
    IsOptional: true;
}
```

However, it's different than the true type answered, which has small "k" `keyArgs`, and small "v" `value`, no `isOptional`:
```ts
{
  keyArgs: [SS58String, FixedSizeBinary<32>]
  value: {
    when: { height: number; index: number }
    deposit: bigint
    depositor: SS58String
    approvals: SS58String[]
  }
}
```

so how do I use the `PaseoQueries['Multisig']['Multisigs']` to type anything?

**Comment by josepot** [comment, marked as answer]
> so how do I use the `PaseoQueries['Multisig']['Multisigs']` to type anything?

The types of a query include the types of the arguments and the types of the returned value, there is also a type that indicates if the return value is optional.

```ts
type MultisigsQuery = PaseQueries['Multisig']['Multisigs']

// If you want to get the types of the arguments:
type MultiSigsArgs = MultisigsQuery['KeyArgs']

// If you want to get the type of the return-value:
type MultiSigsVal = MultisigsQuery['value']

// If you want to get the type of the return-value, but with an `undefined` union when it's optional:
type MultiSigsReturnValue = MultisigsQuery['IsOptional'] extends true
  ? MultisigsVal | undefined
  : MultisigsVal

// If you want to get the type returned by the `getEntries` API, then:
type MultiSigsEntries = Array<{
  keyArgs: MultiSigsArgs,
  value: MultiSigsVal
}>
```

We could add a TS helper to make this simpler, ofc.

**Reply by Tbaut** [reply]
Thank you so much for the super clear answer. Having a helper to get access the following, without having to write it every time would help a lot indeed: 
```ts
interface MultiSigsReturn {
  keyArgs: PaseQueries['Multisig']['Multisigs']['KeyArgs'],
  value: PaseQueries['Multisig']['Multisigs']['Value']
}
```
